.github/workflows/prelease.yml

```diff
@@ -0,0 +1,61 @@
+on:
+  repository_dispatch:
+    types: prelease
+
+jobs:
+  prelease:
+    runs-on: ubuntu-latest
+    name: prelease
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: '${{ github.event.client_payload.branch }}'
+
+      - name: assert on correct commit
+        run: |
+          if [[ "$(git rev-parse HEAD)" != "${{ github.event.client_payload.latest_commit_sha }}" ]]; then
+            echo "Branch ${{ github.event.client_payload.branch }} checked out with SHA $(git rev-parse HEAD) but expected ${{ github.event.client_payload.latest_commit_sha }}"
+            exit 1
+          fi
+
+      - name: git config
+        run: |
+          git config --global user.email "hello@preact.dev"
+          git config --global user.name "prelease"
+
+      - run: npm version ${{ github.event.client_payload.version }}
+
+      - run: npm pack
+
+      - name: Set PRELEASE_GIT_TAG
+        run: echo "::set-env name=PRELEASE_GIT_TAG::$(cat package.json | jq -r '["v",.version]|join("")')"
+
+      - run: git push origin refs/tags/${{ env.PRELEASE_GIT_TAG }}
+
+      - run: git push origin ${{ github.event.client_payload.branch }}
```

Comment:
I'm wondering how this plays out with protected branches. Currently `master` is protected.